### PR TITLE
(BLOG) Porting and Running multi node modules in cloud ci

### PR DESCRIPTION
**Pull request by sheenaajay**
Request for a review

### CODE REVIEW

File: _posts/2021-05-07-running-multinode-modules-gcp.md [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+* Run tests - We use existing serverspec / litmus helpers to setup the dependencies and setup required for module
+  testing.
+* Teardown -  We use provision task to teardown the machines.
+
+Lets go through each step in detail.
+
+***Provision***
+
+Litmus allows the provisioning of a vm/container via a task.
+We use the following [task](https://github.com/puppetlabs/provision/blob/main/tasks/provision_service.json) to provision machine in cloud ci.
```
Comment (carabasdaniel) [inline _posts/2021-05-07-running-multinode-modules-gcp.md:41]:
```suggestion
We use the provision module's [available tasks](https://github.com/puppetlabs/provision/blob/main/tasks) to spin up the test environment.
```

File: _posts/2021-05-07-running-multinode-modules-gcp.md [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+We use the provision module's [available tasks](https://github.com/puppetlabs/provision/blob/main/tasks) to spin up the test environment.
+When run it creates an litmus_inventory.yaml file that allows both [bolt](https://github.com/puppetlabs/bolt) and [serverspec](https://serverspec.org/) to communicate with that vm/container.
+
+How do we differentiate between different vm/containers? 
+Provisioning tasks in the provision module allow us to add arbitrary key/pair values to the bolt inventory file. 
+Using the bolt variables allows users to have multiple labels associated with a single machine.
+
+By running the provision bolt tasks manually or through a bolt plan we can label machines.
+
+![Showing role infm in litmus_inventory.yaml file]({% link /assets/2021-05-07-running-multinode-modules-gcp/inventory_role.png %})
```
Comment (DavidS) [inline _posts/2021-05-07-running-multinode-modules-gcp.md:50]:
```suggestion
![Showing role info in litmus_inventory.yaml file]({% link /assets/2021-05-07-running-multinode-modules-gcp/inventory_role.png %}) 
```

File: _posts/2021-05-07-running-multinode-modules-gcp.md [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+   The below command will only run tests labelled as ‘integration’
+   `bundle exec rake websphere_application_server:integration`
+
+***Teardown***
+
+  We use existing provision task to teardown all the provisioned machines.
+
+## Lets go through different scenarios where we can use multi node testing
+
+***Multilpe puppet agents***
```
Comment (DavidS) [inline _posts/2021-05-07-running-multinode-modules-gcp.md:102]:
```suggestion
***Multiple puppet agents***
```

File: _posts/2021-05-07-running-multinode-modules-gcp.md [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+
+***Run tests***
+
+We use rspec labelling and rake task to identify the tests to run on integeration environment. Lets go through the details.
+
+* Rspec labelling
+  How do we differentiate acceptance tests from integration tests?
+  Using rspec labeling we can label tests, e.g. the test is tagged as integration. Everything else can stay the same.
+
+  ![Example for tagging tests]({% link /assets/2021-05-07-running-multinode-modules-gcp/integration_tag.png %})
```
Comment (DavidS) [inline _posts/2021-05-07-running-multinode-modules-gcp.md:76]: Please change screenshots of code to textual code snippets for accessibility, e.g. that people can copy/paste from there.

File: _posts/2021-05-07-running-multinode-modules-gcp.md [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+we can use bolt plans for provisioning number of nodes with roles tagged to each of it.
+There is new task to install open source puppet server in the provision module.
+[install_puppetserver](https://github.com/puppetlabs/provision/blob/main/tasks/install_puppetserver.json)
+
+Commands
+```
+bundle install --path .bundle/gems/ --jobs 4
+bundle exec rake spec_prep
+bundle exec bolt --modulepath spec/fixtures/modules plan run kubernetes::provision_cluster
+bundle exec bolt --modulepath spec/fixtures/modules plan run kubernetes::puppetserver_setup
```
Comment (DavidS) [inline _posts/2021-05-07-running-multinode-modules-gcp.md:131]: don't these also need `--inventory spec/fixtures/litmus_inventory.yaml` ?

Comment (sheenaajay) [reply _posts/2021-05-07-running-multinode-modules-gcp.md:131]: absolutely. Adding the changes . Thank you

**Comment by sheenaajay** [comment]
Thank you all comments are incorporated.

**Comment by carabasdaniel** [comment]
👍

**Pull request merged**